torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -264,6 +264,59 @@ std::shared_ptr<SugaredValue> ClassValue::attr(
   }
   return std::make_shared<ClassNewMethod>(type_);
 }
+
+std::shared_ptr<SugaredValue> NamedTupleConstructor::call(
+    const SourceRange& loc,
+    Function& m,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    size_t n_binders) {
+  auto nargs = type_->containedTypes().size();
+  if (inputs.size() != nargs) {
+    throw ErrorReport(loc) << "Constructor expected " << nargs << "arguments "
+                           << "but got " << inputs.size();
+  }
+  // TODO: Implement construction with kwargs
+  if (attributes.size() != 0) {
+    throw ErrorReport(loc) << "NamedTuple construction with kwargs is currently"
+                           << " not supported";
+  }
+
+  auto& g = *m.graph();
+
+  // Check that the types of the inputs passed into the NamedTuple constructor
```

Comment:
The thing that generates the FunctionSchema should go in its own function, probably a method of TupleType. It is not great that it is recalculating the FunctionSchema everytime we try to construct a member.